File: ProblemCollection/src/kakao_blind_recruitment2022/ReceiveReportingResults.py
def solution(id_list, report, k):
    report = [i.split() for i in set(report)]
    reported = dict()
    for x, y in report:
        if not reported.get(y):
            reported[y] = set()
        reported[y].add(x)
    for x, y in reported.copy().items():
        if len(y) < k:
            del reported[x]
    return [sum(1 for i in reported.values() if id in i) for id in id_list]
    # answer = []
    # a = list(set(report))
    # dictionary2 = {name : 0 for name in id_list}
    # dictionary = {name : [] for name in id_list}
    # for i in a:
    #     dictionary[i.split()[1]].append(i.split()[0])
    #
    # for i in dictionary:
    #     if len(dictionary[i]) >= k:
    #         for j in dictionary[i]:
    #             dictionary2[j] += 1
    #
    # for i in dictionary2:
    #     answer.append(dictionary2[i])
    #
    # return answer

File: ProblemCollection/src/kakao_blind_recruitment2022/ReceiveReportingResults.js
function solution(id_list, report, k) {
    const answer = {};
    for (let s of report) {
        [x, y] = s.split(' ');
        if (answer[y] == null) {
            answer[y] = new Set();
        }
        answer[y].add(x);
    }
    for (let key of Object.keys(answer)) {
        if (answer[key].size < k) delete answer[key];
    }
    return id_list.map((v) =>
        Object.values(answer).reduce((p, c) => (c.has(v) ? p + 1 : p), 0)
    );
    // let reports = [...new Set(report)].map((a) => {
    //     return a.split(' ');
    // });
    // let counts = new Map();
    // for (const bad of reports) {
    //     counts.set(bad[1], counts.get(bad[1]) + 1 || 1);
    // }
    // let good = new Map();
    // for (const report of reports) {
    //     if (counts.get(report[1]) >= k) {
    //         good.set(report[0], good.get(report[0]) + 1 || 1);
    //     }
    // }
    // let answer = id_list.map((a) => good.get(a) || 0);
    // return answer;
}
